### PR TITLE
feat(pagination-nav): sass module refactor

### LINKED ISSUE #9068: Pagination Nav SASS Refactor
This issue is for updating the Notification to Sass Modules in the `@carbon/styles` package. In this refactor, we will:

- [ ] Copy over styles for for the component to `packages/styles/scss/components/floating-menu/_pagination-nav/scss`
  - Any additional files will live in the folder `packages/styles/scss/components/pagination-nav/`
- [ ] Create an `_index.scss` at `packages/styles/scss/components/pagination-nav/_index.scss`
- [ ] Create a test file at `packages/styles/scss/components/__tests__/pagination-nav/-test.js`

The `_index.scss` should:

- `@forward` the mixin and any variables from `_pagination-nav/.scss`
- Include the styles from `_pagination-nav/.scss` by calling the mixin

The `pagination-nav/-test.js` file should test the following:

- The Public API of the `pagination-nav/_index.scss` file
- Any configuration options that are present for the file

The `_pagination-nav/.scss` will then need to be refactored to use the new import paths and conventions. You can find common recipes here: https://github.com/carbon-design-system/carbon/issues/8899#issuecomment-860887959

If your component includes **component tokens**, follow the instructions below 👇
<details>
  <summary>Component tokens</summary>

- Copy over the `_tokens.scss` file for the component to `scss/components/<component>/_tokens.scss`
- Change any tokens to their new paths, common changes that need to happen will be if the token map uses IDL colors or token colors. These now come from `scss/_colors.scss` and `scss/_theme.scss` specifically.
- Bring in `scss/utilities/_component-tokens.scss` file
- Use this utility to generate four exports in `_tokens.scss`: `$white`, `$g10`, `$g90`, and `$g100`
- Each export should look like the following: `$white: component-tokens.get-tokens($-tokens, themes.$white);`
- Add the tokens to the component exports in `scss/components/<component>/_index.scss` by using `@forward 'tokens'`
- If using storybook, update `packages/carbon-react/.storybook/styles.scss` to include the new component tokens for each theme. This will look like:

```scss
@use '../scss/components/notification';

:root {
  @include styles.theme(themes.$white, notification.$white);
}

.g10 {
  // ...
}
```

You may also need to add the entrypoint in the `@carbon/react` package itself at `packages/carbon-react/scss/components/<component>/_index.scss`
</details>

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #9068 

- Refactors `PaginationNav` to SASS modules 

### Testing: 

- [ ] run `yarn test --watch packages/styles/scss/components/__tests__/pagination-nav-test.js`
- [ ] Make sure tests pass
- [ ] cd `packages/carbon-react` run `yarn storybook` to make sure none of the styles are broken for `PaginationNav`

### CODE REVIEW

File: packages/carbon-react/src/components/PaginationNav/PaginationNav.stories.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+_PaginationNav.storyName = 'PaginationNav';
+
+_PaginationNav.parameters = {
+  info: {
+    text: `
+         Pagination Nav is a group of pagination buttons.
+           `,
+  },
+};
```
Comment (tw15egan) [inline packages/carbon-react/src/components/PaginationNav/PaginationNav.stories.js]: I think we can just get rid of this stuff 👍🏻

File: packages/styles/scss/components/pagination-nav/_pagination-nav.scss [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    }
+  }
+
+  .#{$prefix}--pagination-nav__page--active + &::after,
+  &.#{$prefix}--pagination-nav__page--active::after {
+    left: calc(50% - #{$spacing-05 / 2});
+    width: $spacing-05;
+    opacity: 1;
+  }
+}
```
Comment (tw15egan) [inline packages/styles/scss/components/pagination-nav/_pagination-nav.scss]: Would it make sense to have this in its own `mixins.scss` file?

Comment (andreancardona) [reply packages/styles/scss/components/pagination-nav/_pagination-nav.scss]: The current `pagination-nav` doesn't have one... 🤷🏽‍♀️

Comment (andreancardona) [reply packages/styles/scss/components/pagination-nav/_pagination-nav.scss]: just updated it with a new `_mixins` file :)

File: packages/carbon-react/src/components/PaginationNav/PaginationNav.stories.js (hunk 1 above)
Comment (andreancardona) [reply packages/carbon-react/src/components/PaginationNav/PaginationNav.stories.js]: done!

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by sstrubberg — APPROVED** [review]

**Pull request merged**